holoviews/core/data/__init__.py

```diff
@@ -132,7 +133,15 @@ def __call__(self, new_type, kdims=None, vdims=None, groupby=None,
             else:
                 selected = self._element
         else:
-            selected = self._element.reindex(groupby+kdims, vdims)
+            if pd and issubclass(self._element.interface, PandasInterface):
+                ds_dims = self._element.dimensions()
+                ds_kdims = [self._element.get_dimension(d) if d in ds_dims else d
+                            for d in groupby+kdims]
+                ds_vdims = [self._element.get_dimension(d) if d in ds_dims else d
+                            for d in vdims]
+                selected = self._element.clone(kdims=ds_kdims, vdims=ds_vdims)
+            else:
+                selected = self._element.reindex(groupby+kdims, vdims)
```

Comment:
Is the same logic used elsewhere in a context other than the conversion interface? Is any code shared and is it worth abstracting out?

Reply:
No, I cannot think of anywhere this could be reused atm.